src/transformStateWithClones.js

```diff
@@ -7,7 +7,27 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const results = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        break;
+      case 'addProperties':
+        currentState = { ...currentState, ...action.extraData };
+        break;
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete currentState[key];
+        }
+        break;
+    }
+    results.push({ ...currentState });
```

Comment:
Instead of pushing the copy of the state inside the `switch` block, it's better to do it once after the `switch` statement. This will adhere to the DRY (Don't Repeat Yourself) principle and ensure that the copying of the state is done in one place.